untitled1/src/main/com/aca/homework/week8/product/repository/InMemoryProductRepository.java

```diff
@@ -0,0 +1,42 @@
+package com.aca.homework.week8.product.repository;
+
+public class InMemoryProductRepository implements ProductRepository {
+    private final int size = 10000;
+    private Product[] products = new Product[size];
+    private int counter;
+
+    @Override
+    public void save(Product product) {
+        if (product == null)
+            throw new IllegalArgumentException("product is null");
+        Product elementFound = findProduct(product);
+        if (elementFound != null)
+            throw new ProductAlreadyExistsException("element exists");
+        products[counter++] = product;
+    }
+
+    private Product findProduct(Product product) {
+        for (int i = 0; i < counter; i++)
+            if (products[i].equals(product))
+                return products[i];
+        return null;
+    }
+
+    @Override
+    public Product findById(Long id) {
+        for (int i = 0; i < counter; i++)
+            if (products[i].getId() == id) return products[i];
+        return null;
+    }
+
+    @Override
+    public Product getById(Long Id) {
+        for (int i = 0; i < counter; i++)
+            if (products[i].getId() == Id) return products[i];
+        throw new ProductNotFoundException("element don't find");
```

Comment:
also please print the id for which the product is not found. would be better to pass id to the exception constructor and build the message there